cdap-client/src/main/java/co/cask/cdap/client/ScheduleClient.java

```diff
@@ -70,6 +74,49 @@ public ScheduleClient(ClientConfig config) {
   }
 
   @Deprecated
+  public void add(Id.Schedule schedule, ScheduleInstanceConfiguration configuration) throws IOException,
+    UnauthenticatedException, NotFoundException, UnauthorizedException, AlreadyExistsException {
+    add(schedule.toEntityId(), configuration);
+  }
+
+  public void add(ScheduleId scheduleId, ScheduleInstanceConfiguration configuration) throws IOException,
+    UnauthenticatedException, NotFoundException, UnauthorizedException, AlreadyExistsException {
+
+    String path = String.format("apps/%s/versions/%s/schedules/%s",
+                                scheduleId.getApplication(), scheduleId.getVersion(), scheduleId.getSchedule());
```

Comment:
Currently, if you build a `ScheduleId` with no version, the version will default to `ApplicationId.DEFAULT_VERSION`. Additionally, if you call the unversioned API, internally it will use `ApplicationId.DEFAULT_VERSION` as well.